### DISCUSSION TITLE
Database locked

### DISCUSSION THREAD

**Discussion by Abbos27** [simolus3/drift#3073] [Q&A]
What can cause a database file to be locked on Windows? On some Windows devices, the database can occasionally become locked when trying to establish a connection, but it is not locked permanently

**Comment by frankvollebregt** [comment]
A common reason is that an existing connection is currently accessing the database. Maybe you can try around with the Write-Ahead Log (WAL): https://www.sqlite.org/wal.html

There are some considerations on using it in the Drift docs, though: https://drift.simonbinder.eu/docs/examples/existing_databases/#additional-considerations-for-wal

This was brought to my attention by @simolus3 when I previously experienced some trouble while working with isolates (on mobile): https://github.com/simolus3/drift/discussions/2957#discussioncomment-9134206

Since it's Windows (purely anecdotally speaking, I have no idea if any of this is realistic), I wouldn't be surprised if an antivirus software would also be snooping around from time to time to check if nothing strange is going on: if you haven't you might be able to add your program folder/files as an exception there too?
